sources/metricdog/src/systemd.rs

```diff
@@ -0,0 +1,270 @@
+
+    #[test]
+    fn parse_stdout_exit_0() {
+        let got = parse_property(
+            format!("{}=0", EXIT_STATUS_PROPERTY).as_str(),
+            EXIT_STATUS_PROPERTY,
+        )
+        .map(|exit_code| exit_code.parse::<i32>().ok())
+        .flatten()
+        .unwrap();
+
+        let want = 0;
+        assert_eq!(got, want);
+    }
+
+    #[test]
+    fn parse_stdout_exit_255() {
+        let got = parse_property(
+            format!("{}=255", EXIT_STATUS_PROPERTY).as_str(),
+            EXIT_STATUS_PROPERTY,
+        )
+        .map(|exit_code| exit_code.parse::<i32>().ok())
+        .flatten()
+        .unwrap();
+
+        let want = 255;
+        assert_eq!(got, want);
+    }
+
+    #[test]
+    fn parse_stdout_exit_0_with_newline() {
+        let got = parse_property(
+            format!("{}=0\n", EXIT_STATUS_PROPERTY).as_str(),
+            EXIT_STATUS_PROPERTY,
+        )
+        .map(|exit_code| exit_code.parse::<i32>().ok())
+        .flatten()
+        .unwrap();
+
+        let want = 0;
+        assert_eq!(got, want);
+    }
+
+    #[test]
+    fn parse_stdout_exit_255_with_newline() {
+        let got = parse_property(
+            format!("{}=255\n", EXIT_STATUS_PROPERTY).as_str(),
+            EXIT_STATUS_PROPERTY,
+        )
+        .map(|exit_code| exit_code.parse::<i32>().ok())
+        .flatten()
+        .unwrap();
+
+        let want = 255;
+        assert_eq!(got, want);
+    }
+
+    #[test]
+    fn parse_stdout_exit_extra_chars() {
+        let got = parse_property(
+            format!("{}=255foo", EXIT_STATUS_PROPERTY).as_str(),
+            EXIT_STATUS_PROPERTY,
+        )
+        .map(|exit_code| exit_code.parse::<i32>().ok())
+        .flatten();
+
+        assert!(got.is_none());
+    }
+
+    #[test]
+    fn parse_stdout_malformed() {
+        let got = parse_property(
+            format!("{} = 123", EXIT_STATUS_PROPERTY).as_str(),
+            EXIT_STATUS_PROPERTY,
+        )
+        .map(|exit_code| exit_code.parse::<i32>().ok())
+        .flatten();
+
+        assert!(got.is_none());
+    }
+
+    #[test]
+    fn parse_stdout_empty_string() {
+        let got = parse_property("", EXIT_STATUS_PROPERTY)
+            .map(|exit_code| exit_code.parse::<i32>().ok())
+            .flatten();
+
+        assert!(got.is_none());
+    }
+
+    #[test]
+    fn parse_stdout_property_only() {
+        let got = parse_property(EXIT_STATUS_PROPERTY, EXIT_STATUS_PROPERTY)
+            .map(|exit_code| exit_code.parse::<i32>().ok())
+            .flatten();
+
+        assert!(got.is_none());
+    }
+
+    #[test]
+    fn parse_stdout_property_and_equals_only() {
+        let got = parse_property(
+            format!("{}=", EXIT_STATUS_PROPERTY).as_str(),
+            EXIT_STATUS_PROPERTY,
+        )
+        .map(|exit_code| exit_code.parse::<i32>().ok())
+        .flatten();
+
+        assert!(got.is_none());
+    }
+}
+
+impl HostCheck for SystemdCheck {
```

Comment:
nit: I'd put this before the `parse_property` function or just before the tests.